File: TravelAgencyRestApi/src/main/java/com/travelagency/travelagencyapplication/dto/request/ActivityRequest.java
package com.travelagency.travelagencyapplication.dto.request;

import com.travelagency.travelagencyapplication.collection.User;
import com.travelagency.travelagencyapplication.enums.ActivityCostType;
import com.travelagency.travelagencyapplication.enums.ActivityType;
import com.travelagency.travelagencyapplication.enums.PaymentType;
import lombok.*;
import org.springframework.web.multipart.MultipartFile;

import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ActivityRequest{

    private String id;
    private String title;
    private String destination;
    private Set<MultipartFile> imagesFile;
    private Set<String> imagesUrl;
    private String shortDescription;
    private String longDescription;

    private String activityGuideId;
    private ActivityType activityType;
    private ActivityCostType activityCostType;

    private Double activityAdditionalPrice;

}

File: TravelAgencyFrontend/src/app/types/Activity/ActivityRequest.ts
import {ActivityType} from "./ActivityType";
import {ActivityCostType} from "./ActivityCostType";

export interface ActivityRequest {
  id?: string;
  title: string;
  destination: string;
  imagesFile: File[];
  shortDescription: string;
  longDescription: string;
  activityGuideId: string;
  activityType: ActivityType;
  activityCostType: ActivityCostType;
  activityAdditionalPrice?: number; // Optional field
}
